project/driver_controls/src/input_interrupt.c

```diff
@@ -0,0 +1,150 @@
+#include "input_interrupt.h"
+#include "input_event.h"
+#include "event_queue.h"
+#include "driver_io.h"
+
+// Number of available GPIO interrupt channels
+#define NUM_INPUT_EVENTS 16
+
+// Arbitrary thresholds for gas pedal
+#define COAST_THRESHOLD 1000
+#define DRIVE_THRESHOLD 3000
+
+// Driver IO pin definitions
+#define DRIVER_PIN_0  0
```

Comment:
Those macros were supposed to allow me to get rid of the magic pin numbers. They've been replaced with enums, however.